src/composables/auth/useFirebaseAuthActions.ts

```diff
@@ -145,12 +145,32 @@ export const useFirebaseAuthActions = () => {
     })
   }, reportError)
 
+  const fetchSubscriptionStatus = wrapWithErrorHandlingAsync(async () => {
+    return await authStore.fetchSubscriptionStatus()
+  }, reportError)
+
+  const subscribeToCloud = wrapWithErrorHandlingAsync(async () => {
+    const response = await authStore.initiateSubscriptionCheckout()
+
+    if (!response.checkout_url) {
+      throw new Error(
+        t('toastMessages.failedToInitiateSubscription', {
+          error: 'No checkout URL returned'
+        })
+      )
+    }
+
+    window.open(response.checkout_url, '_blank')
+  }, reportError)
+
```

Comment:
We should pull these out into `useSubscription`. It's already importing the `authStore`.